verification/cbmc/proofs/sign_header/sign_header_harness.c

```diff
@@ -49,6 +49,11 @@ void sign_header_harness() {
         session_setup(MAX_TABLE_SIZE, MAX_TRACE_LIST_ITEMS, MAX_EDK_LIST_ITEMS, MAX_BUFFER_SIZE, MAX_STRING_LEN);
 
     /* Assumptions */
+    if (session->alg_props == NULL) {
+        struct aws_cryptosdk_alg_properties *props = ensure_alg_properties_attempt_allocation(MAX_STRING_LEN);
+        __CPROVER_assume(aws_cryptosdk_alg_properties_is_valid(props));
```

Comment:
I think this precondition also needs to be added to the `sign_header` function.

Reply:
I think it has already been added. You mean [this one](https://github.com/aws/aws-encryption-sdk-c/pull/661/files#diff-1b12f74b2bb9289eb9758d7786c2ac0c348ec714223f4725b976e883d3d39d75R176-R179), right?

Reply:
Ah, you're right. My bad.